File: antia_numerical_methods/c/fun/betai.c
/*	To calculate the incomplete Beta function I_x(a,b) using
	the integral, called by betap

	a,b : (input) Arguments for the complete Beta function
	x : (input) Upper limit of integration defining the incomplete
			Beta function

	Required routines : gammaln, adpint, kronrd, fbeta
*/

#include <math.h>

	double fbeta(double x);
	int adpint(double *rint, double xl, double xu, double reps, double aeps,
		double *dif, double f1(double x), int *n, int *nmax);
	int kronrd(double *ri, double a, double b, double *dif, int *n,
		double f1(double x));
	double AA, BB;

double betai(double a, double b, double x)
{
	double xl,xu,reps,aeps,b1,rint,dif,f;
	int i,npt,nmax;

	AA=a; BB=b; xl=0; xu=x; reps=1e-14; aeps=1e-280;
	i=adpint(&rint,xl,xu,reps,aeps,&dif,fbeta,&npt,&nmax);
	b1=log(rint)+gammaln(a+b)-gammaln(a)-gammaln(b);
	f=exp(b1);
	if(f>1) f=1;
	return f;
}
	

File: antia_numerical_methods@Beliavsky/f77/sub16/betai.f
!	To calculate the incomplete Beta function I_x(a,b) using
!	the integral, called by BETAP
!
!	A,B : (input) Arguments for the complete Beta function
!	X : (input) Upper limit of integration defining the incomplete
!                       Beta function
!
!	Required routines : GAMMAL, ADPINT, KRONRD, FBETA

      FUNCTION BETAI(A,B,X)
      IMPLICIT REAL*16(A-H,O-Z)
      EXTERNAL FBETA
      COMMON/BETA/AA,BB

      AA=A
      BB=B
      XL=0
      XU=X
      REPS=1.Q-14
      AEPS=1.Q-280
      CALL ADPINT(RINT,XL,XU,REPS,AEPS,DIF,FBETA,IER,NPT,NMAX)
      B1=LOG(RINT)+GAMMAL(A+B)-GAMMAL(A)-GAMMAL(B)
      BETAI=EXP(B1)
      IF(BETAI.GT.1.0) BETAI=1.0
      END
